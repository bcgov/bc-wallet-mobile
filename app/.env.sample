MEDIATOR_URL=https://f326-207-194-65-204.ngrok.io?c_i=eyJAdHlwZSI6ICJodHRwczovL2RpZGNvbW0ub3JnL2Nvbm5lY3Rpb25zLzEuMC9pbnZpdGF0aW9uIiwgIkBpZCI6ICI2MjQ0ZThiNS0wNWYzLTRhYWItYjM1Yy1lYWVlMWNmZTAyM2MiLCAicmVjaXBpZW50S2V5cyI6IFsiQ3lqM1BHRUJzQ3RyUGFtTTQyRngza3BlYmR2QWdNd1lGejlFS3RmNnlUN3giXSwgImxhYmVsIjogIk1lZGlhdG9yIiwgInNlcnZpY2VFbmRwb2ludCI6ICJodHRwczovL2YzMjYtMjA3LTE5NC02NS0yMDQubmdyb2suaW8ifQ==

# Push notification variables
MEDIATOR_USE_PUSH_NOTIFICATIONS=false
MEDIATOR_LABEL=Mediator