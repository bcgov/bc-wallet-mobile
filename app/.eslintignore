node_modules
__tests__
__mocks__
ios
android

# generated by bob
lib/
bifold-app/
generate-app-package-json.js