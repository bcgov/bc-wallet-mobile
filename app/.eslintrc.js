module.exports = {
  root: true,
  parser: '@typescript-eslint/parser',
  parserOptions: {
    tsconfigRootDir: __dirname,
    ecmaVersion: 2020, // Allows for the parsing of modern ECMAScript features
    sourceType: 'module', // Allows for the use of imports
  },
  extends: [
    'plugin:import/errors',
    'plugin:import/warnings',
    'plugin:import/typescript',
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended', // Uses the recommended rules from the @typescript-eslint/eslint-plugin
    'plugin:prettier/recommended', // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
  ],
  settings: {
    // 'import/ignore': ['react-native'],
    // 'import/extensions': ['.js', '.ts', 'tsx', 'jsx'],
    // 'import/parsers': {
    //   '@typescript-eslint/parser': ['.ts', '.tsx'],
    // },
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {},
    },
  },
  plugins: ['@typescript-eslint', 'import'],
  rules: {
    // Type is enforced by callers. Not entirely, but it's good enough.
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'off',
    // Aries protocol defines attributes with snake case.
    '@typescript-eslint/camelcase': 'off',
    '@typescript-eslint/no-use-before-define': ['error', { functions: false, classes: false, variables: true }],
    '@typescript-eslint/explicit-member-accessibility': 'error',
    // Because of early development, we only warn on ts-ignore. In future we want to move to error
    '@typescript-eslint/ban-ts-comment': 'warn',
    'no-console': 'error',
    'import/no-cycle': 'error',
    'import/order': [
      'error',
      {
        groups: ['type', ['builtin', 'external'], 'parent', 'sibling', 'index'],
        alphabetize: {
          order: 'asc',
        },
        'newlines-between': 'always',
      },
    ],
    'import/no-extraneous-dependencies': [
      'error',
      {
        devDependencies: false,
      },
    ],
  },
  globals: {
    require: true,
  },
  overrides: [
    {
      files: ['*.test.*'],
      rules: {
        'no-console': 'off',
        'import/no-extraneous-dependencies': 'off',
      },
    },
    {
      files: ['aries-components/**/*.ts'],
      rules: {
        '@typescript-eslint/no-namespace': 'off',
      },
    },
    {
      files: ['.eslintrc.js', '*.config.js'],
      env: {
        node: true,
      },
    },
  ],
}
