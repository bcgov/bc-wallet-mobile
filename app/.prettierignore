package-lock.json
app.json
ios/
android/
node_modules/
generate-app-package-json.js
src/configs/ledgers/indy/ledgers.json
.gitignore
