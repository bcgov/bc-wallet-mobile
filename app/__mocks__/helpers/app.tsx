import 'reflect-metadata'
import React, { PropsWithChildren, useMemo } from 'react'

import { container } from 'tsyringe'
import { MainContainer, ContainerProvider } from '@hyperledger/aries-bifold-core'

export const BasicAppContext: React.FC<PropsWithChildren> = ({ children }) => {
  const context = useMemo(() => new MainContainer(container.createChildContainer()).init(), [])
  return (
    <ContainerProvider value={context}>{children}</ContainerProvider>
  )
}