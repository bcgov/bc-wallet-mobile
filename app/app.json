{
  "name": "BCWallet",
  "displayName": "BC Wallet"
}
