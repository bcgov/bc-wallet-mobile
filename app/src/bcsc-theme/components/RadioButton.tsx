import { useTheme } from '@bifold/core'
import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'

export interface RadioButtonProps {
  label: string
  value: string
  selected: boolean
  onPress: (value: string) => void
  disabled?: boolean
  testID?: string
}

export const RadioButton: React.FC<RadioButtonProps> = ({
  label,
  value,
  selected,
  onPress,
  disabled = false,
  testID,
}) => {
  const { ColorPalette, TextTheme, Spacing } = useTheme()

  const styles = StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      paddingVertical: Spacing.md,
      paddingHorizontal: Spacing.lg,
      backgroundColor: ColorPalette.brand.secondaryBackground,
    },
    radioCircle: {
      width: Spacing.lg,
      height: Spacing.lg,
      borderRadius: Spacing.lg / 2,
      borderWidth: 3,
      borderColor: disabled ? ColorPalette.grayscale.mediumGrey : ColorPalette.brand.primary,
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: 'transparent',
    },
    innerCircle: {
      width: Spacing.lg / 2,
      height: Spacing.lg / 2,
      borderRadius: Spacing.lg / 4,
      backgroundColor: disabled ? ColorPalette.grayscale.mediumGrey : ColorPalette.brand.primary,
    },
    label: {
      ...TextTheme.normal,
      color: disabled ? ColorPalette.grayscale.mediumGrey : ColorPalette.grayscale.white,
      flex: 1,
    },
  })

  const handlePress = () => {
    if (!disabled) {
      onPress(value)
    }
  }

  return (
    <TouchableOpacity
      style={styles.container}
      onPress={handlePress}
      disabled={disabled}
      accessible
      accessibilityRole="radio"
      accessibilityState={{ selected, disabled }}
      accessibilityLabel={label}
      testID={testID}
    >
      <Text style={styles.label}>{label}</Text>
      <View style={styles.radioCircle}>{selected && <View style={styles.innerCircle} />}</View>
    </TouchableOpacity>
  )
}
