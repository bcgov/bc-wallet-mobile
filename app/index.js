/* eslint-disable import/no-extraneous-dependencies */
/**
 * @format
 */
import 'react-native-gesture-handler'
// remove these when updated to react-native 0.65.0
import '@formatjs/intl-getcanonicallocales/polyfill'
import '@formatjs/intl-locale/polyfill'
import '@formatjs/intl-pluralrules/polyfill'
import '@formatjs/intl-pluralrules/locale-data/en' // locale-data for en
import '@formatjs/intl-displaynames/polyfill'
import '@formatjs/intl-displaynames/locale-data/en' // locale-data for en
import '@formatjs/intl-listformat/polyfill'
import '@formatjs/intl-listformat/locale-data/en' // locale-data for en
import '@formatjs/intl-numberformat/polyfill'
import '@formatjs/intl-numberformat/locale-data/en' // locale-data for en
import '@formatjs/intl-relativetimeformat/polyfill'
import '@formatjs/intl-relativetimeformat/locale-data/en' // locale-data for en
import '@formatjs/intl-datetimeformat/polyfill'
import '@formatjs/intl-datetimeformat/locale-data/en' // locale-data for en
import '@formatjs/intl-datetimeformat/add-all-tz' // Add ALL tz data

import { NavigationContainer } from '@react-navigation/native'
import React from 'react'
import { AppRegistry } from 'react-native'

import App from './App'
import { name as appName } from './app.json'
import bcwallet from './src/'
const { theme } = bcwallet

const { ColorPallet } = theme

const navigationTheme = {
  dark: true,
  colors: {
    primary: ColorPallet.brand.primary,
    background: ColorPallet.brand.primaryBackground,
    card: ColorPallet.brand.primary,
    text: ColorPallet.grayscale.white,
    border: ColorPallet.grayscale.white,
    notification: ColorPallet.grayscale.white,
  },
}

const Base = () => {
  return (
    <NavigationContainer theme={navigationTheme}>
      <App />
    </NavigationContainer>
  )
}
AppRegistry.registerComponent(appName, () => Base)
