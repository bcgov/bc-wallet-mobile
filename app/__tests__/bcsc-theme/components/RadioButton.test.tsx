import { fireEvent, render, waitFor } from '@testing-library/react-native'
import React from 'react'

import { BasicAppContext } from '../../../__mocks__/helpers/app'
import { RadioButton } from '../../../src/bcsc-theme/components/RadioButton'

describe('RadioButton Component', () => {
  const defaultProps = {
    label: 'Test Option',
    value: 'test-value',
    selected: false,
    onPress: jest.fn(),
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('Rendering', () => {
    test('renders correctly when unselected', () => {
      const tree = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('renders correctly when selected', () => {
      const tree = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} selected={true} />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('renders correctly when disabled', () => {
      const tree = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} disabled={true} />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('renders correctly when selected and disabled', () => {
      const tree = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} selected={true} disabled={true} />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('displays the correct label text', () => {
      const { getByText } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} label="Custom Label" />
        </BasicAppContext>
      )

      expect(getByText('Custom Label')).toBeTruthy()
    })
  })

  describe('Interaction', () => {
    test('calls onPress with correct value when pressed', () => {
      const mockOnPress = jest.fn()
      const { getByRole } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} onPress={mockOnPress} />
        </BasicAppContext>
      )

      const radioButton = getByRole('radio')
      fireEvent.press(radioButton)

      expect(mockOnPress).toHaveBeenCalledWith('test-value')
      expect(mockOnPress).toHaveBeenCalledTimes(1)
    })

    test('does not call onPress when disabled', () => {
      const mockOnPress = jest.fn()
      const { getByRole } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} onPress={mockOnPress} disabled={true} />
        </BasicAppContext>
      )

      const radioButton = getByRole('radio')
      fireEvent.press(radioButton)

      expect(mockOnPress).not.toHaveBeenCalled()
    })

    test('calls onPress multiple times for multiple presses', () => {
      const mockOnPress = jest.fn()
      const { getByRole } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} onPress={mockOnPress} />
        </BasicAppContext>
      )

      const radioButton = getByRole('radio')
      fireEvent.press(radioButton)
      fireEvent.press(radioButton)
      fireEvent.press(radioButton)

      expect(mockOnPress).toHaveBeenCalledTimes(3)
    })
  })

  describe('Accessibility', () => {
    test('has correct accessibility role', () => {
      const { getByRole } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} />
        </BasicAppContext>
      )

      expect(getByRole('radio')).toBeTruthy()
    })

    test('has correct accessibility state when unselected', () => {
      const { getByRole } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} selected={false} />
        </BasicAppContext>
      )

      const radioButton = getByRole('radio')
      expect(radioButton.props.accessibilityState.selected).toBe(false)
    })

    test('has correct accessibility state when selected', () => {
      const { getByRole } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} selected={true} />
        </BasicAppContext>
      )

      const radioButton = getByRole('radio')
      expect(radioButton.props.accessibilityState.selected).toBe(true)
    })

    test('has correct accessibility state when disabled', () => {
      const { getByRole } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} disabled={true} />
        </BasicAppContext>
      )

      const radioButton = getByRole('radio')
      expect(radioButton.props.accessibilityState.disabled).toBe(true)
    })

    test('has correct accessibility label', () => {
      const { getByRole } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} label="Accessibility Label" />
        </BasicAppContext>
      )

      const radioButton = getByRole('radio')
      expect(radioButton.props.accessibilityLabel).toBe('Accessibility Label')
    })

    test('has correct testID when provided', () => {
      const { getByTestId } = render(
        <BasicAppContext>
          <RadioButton {...defaultProps} testID="custom-test-id" />
        </BasicAppContext>
      )

      expect(getByTestId('custom-test-id')).toBeTruthy()
    })
  })
})
