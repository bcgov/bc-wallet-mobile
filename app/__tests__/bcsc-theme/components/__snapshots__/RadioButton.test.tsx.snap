// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component Rendering renders correctly when disabled 1`] = `
<View
  accessibilityLabel="Test Option"
  accessibilityRole="radio"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": false,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#606060",
        "flex": 1,
        "fontFamily": "BCSans-Regular",
        "fontSize": 18,
        "fontWeight": "normal",
      }
    }
  >
    Test Option
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#606060",
        "borderRadius": 12,
        "borderWidth": 3,
        "height": 24,
        "justifyContent": "center",
        "width": 24,
      }
    }
  />
</View>
`;

exports[`RadioButton Component Rendering renders correctly when selected 1`] = `
<View
  accessibilityLabel="Test Option"
  accessibilityRole="radio"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": true,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "flex": 1,
        "fontFamily": "BCSans-Regular",
        "fontSize": 18,
        "fontWeight": "normal",
      }
    }
  >
    Test Option
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#003366",
        "borderRadius": 12,
        "borderWidth": 3,
        "height": 24,
        "justifyContent": "center",
        "width": 24,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#003366",
          "borderRadius": 6,
          "height": 12,
          "width": 12,
        }
      }
    />
  </View>
</View>
`;

exports[`RadioButton Component Rendering renders correctly when selected and disabled 1`] = `
<View
  accessibilityLabel="Test Option"
  accessibilityRole="radio"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": true,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#606060",
        "flex": 1,
        "fontFamily": "BCSans-Regular",
        "fontSize": 18,
        "fontWeight": "normal",
      }
    }
  >
    Test Option
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#606060",
        "borderRadius": 12,
        "borderWidth": 3,
        "height": 24,
        "justifyContent": "center",
        "width": 24,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#606060",
          "borderRadius": 6,
          "height": 12,
          "width": 12,
        }
      }
    />
  </View>
</View>
`;

exports[`RadioButton Component Rendering renders correctly when unselected 1`] = `
<View
  accessibilityLabel="Test Option"
  accessibilityRole="radio"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": false,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "flex": 1,
        "fontFamily": "BCSans-Regular",
        "fontSize": 18,
        "fontWeight": "normal",
      }
    }
  >
    Test Option
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#003366",
        "borderRadius": 12,
        "borderWidth": 3,
        "height": 24,
        "justifyContent": "center",
        "width": 24,
      }
    }
  />
</View>
`;
